docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,567 @@
+   but with `libpython2-dev` instead of `libpython-dev`.
+
+   **Note:** LLDB currently requires at least `swig-1.3.40` but will
+   successfully build with version 2 shipped with Ubuntu.
+
+## Building the project for the first time
+
+### Spot check dependencies
+
+* Run `cmake --version`: This should be 3.18.1 or higher for macOS.
+* Run `python3 --version`: Check that this succeeds.
+* Run `ninja --version`: Check that this succeeds.
+* Run `sccache --version`: Check that this succeeds.
+
+### The roles of different tools
+
+At this point, it is worthwhile to pause for a moment
+to understand what the different tools do:
+
+1. CMake is a cross-platform build system for C and C++.
+   It forms the core infrastructure used to _configure_ builds of
+   Swift and its companion projects.
+2. Ninja is a low-level build system that can be used to _build_ the project,
+   as an alternative to Xcode's build system. Ninja is somewhat faster,
+   especially for incremental builds, and supports more build environments.
+3. Sccache is a caching tool:
+   If you ever delete your build directory and rebuild from scratch
+   (i.e. do a "clean build"), Sccache can accelerate the new build
+   significantly. There are few things more satisfying than seeing Sccache
+   cut through build times.
+4. `utils/update-checkout` is a script to help you work with all the individual
+   git repositories together, instead of manually cloning/updating each one.
+5. `utils/build-script` (we will introduce this shortly)
+   is a high-level automation script that handles configuration (via CMake),
+   building (via Ninja or Xcode), caching (via Sccache), running tests and more.
+
+> **Pro Tip**: Most tools support `--help` flags describing the options they
+> support. Additionally, both Clang and the Swift compiler have hidden flags
+> (`clang --help-hidden`/`swiftc --help-hidden`) and frontend flags
+> (`clang -cc1 --help`/`swiftc -frontend --help`) and the Swift compiler
+> even has hidden frontend flags (`swiftc -frontend --help-hidden`). Sneaky!
+
+Phew, that's a lot to digest! Now let's proceed to the actual build itself!
+
+### The actual build
+
+1. Make sure you have Sccache running.
+   ```sh
+   sccache --start-server
+   ```
+   (Optional) Sccache defaults to a cache size of 10GB, which is relatively
+   small compared to build artifacts. You can bump it up, say by setting
+   `export SCCACHE_CACHE_SIZE="50G"` in your dotfile(s). For more details,
+   see the [Sccache README][Sccache].
+2. Decide if you would like to build the toolchain using Ninja or using Xcode.
+   - If you use an editor other than Xcode and/or you want somewhat faster builds,
+     go with Ninja.
+   - If you are comfortable with using Xcode and would prefer to use it,
+     go with Xcode.
+3. Build the toolchain with optimizations, debuginfo, and assertions and run
+   the tests.
+   - Via Ninja:
+     ```sh
+     utils/build-script --skip-build-benchmark \
```

Comment:
@varungandhi-apple The correct flag wouldn't be `--skip-build-benchmarks`? I just ran `build-script --help` and seems it is. Or they both work?

Reply:
Good point. Turns out, `build-script` does not check if any argument is unused, so I've been passing it `--skip-build-benchmark` for several months now and it just seems to be ignoring it.

Reply:
Right, so maybe this is something that could at least warn because typos are very common and we may end up as you mentioned using a flag that is silently being ignored. Should we open an SR for this?

Reply:
Yes please, a bug report would be great. I will add a warning to the "Troubleshooting build issues" section about typos.

Reply:
@varungandhi-apple SR created https://bugs.swift.org/browse/SR-13494 :)

Reply:
Thanks, I've added a warning about typos.